src/ZoneServer/Skills/Handlers/Swordsmen/Barbarian/Barbarian_GiantSwing.cs

```diff
@@ -0,0 +1,263 @@
+			{
+				modifier.DamageMultiplier += 0.12f * wildNature.OverbuffCounter;
+			}
+
+			var skillHitResult = SCR_SkillHit(caster, target, skill, modifier);
+			target.TakeDamage(skillHitResult.Damage, caster);
+
+			var skillHit = new SkillHitInfo(caster, target, skill, skillHitResult, damageDelay, skillHitDelay);
+			skillHit.KnockBackInfo = new KnockBackInfo(caster.Position, target.Position, skill);
+			skillHit.HitInfo.Type = skill.Data.KnockDownHitType;
+			target.Position = skillHit.KnockBackInfo.ToPosition;
+			hits.Add(skillHit);
+
+
+			Send.ZC_SKILL_HIT_INFO(caster, hits);
+
+			await Task.Delay(tossDelay);
+			cancelToken.Cancel();
+
+			// this skill's animation doesn't end unless you cancel it
+			Send.ZC_NORMAL.SpinObject(caster, 0, 0, 0, 0);
+			Send.ZC_PLAY_ANI(caster, "idle1");
+			caster.TurnTowards(chainDirection);
+			Send.ZC_NORMAL.SkillCancelCancel(caster, skill.Id);
+
+
+			target.StartBuff(BuffId.giantswing_Debuff, skill.Level, 0, TimeSpan.FromSeconds(10), caster);
+
+			var targetPos = chainPos.GetRelative(chainDirection, TossDistance);
+			targetPos = caster.Map.Ground.GetLastValidPosition(target.Position, targetPos);
+
+			// have to stop the target before the leapjump or their position can get messed up
+			if (target.Components.TryGet<MovementComponent>(out var movementComponent))
+				movementComponent.Stop();
+
+			target.Position = targetPos;
+			Send.ZC_NORMAL.LeapJump(target, targetPos, 0.1f, 0.1f, 1f, 0.2f, 1f, 30);
+		}
+
+
+		/// <summary>
+		/// Spins the target around.
+		/// </summary>
+		/// <remarks>
+		/// This is not the correct implementation, it should call something like
+		/// Send.ZC_ATTACH_TO_OBJ(target, null, "ChainTest", "Bone_chain13", 0.01f, 1, 1, "", 0, 0, 0);
+		/// However the correct values to make this function have not been found
+		/// </remarks>
+		/// <param name="caster"></param>
+		/// <param name="target"></param>
+		/// <param name="duration"></param>
+		private async Task SpinTarget(ICombatEntity caster, ICombatEntity target, CancellationToken cancelToken)
+		{
+			var tickTime = TimeSpan.FromMilliseconds(70);
+
+			while (!cancelToken.IsCancellationRequested)
+			{
+				target.Position = new Position(caster.Position.GetRelative(caster.Direction, ChainLength));
+				Send.ZC_SET_POS(target);
+				Send.ZC_NORMAL.PlayEffect(target, "F_burstup022_smoke", 0.7f);
+
+				await Task.Delay(tickTime);
+			}
+		}
+
+
+		/// <summary>
+		/// Deals damage to other targets in the range of the spin
+		/// </summary>
+		/// <remarks>
+		/// This is custom functionality, the original skill did not
+		/// hit additional targets, but it definitely feels like it
+		/// should have done something like this
+		/// </remarks>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async Task SpinAttack(Skill skill, ICombatEntity caster, ICombatEntity spinTarget, ISplashArea splashArea)
```

Comment:
I like that you're experimenting with the skills and want to leave sensible custom functionality in the code, but I feel like its usability is limited in this case if there isn't even a simple way for users to enable it, aside from digging through the code. Meanwhile, it would have to be maintained though, in case anything about the main code is changed.

How would you feel about us creating a more accessible way to do such customizations? For example, while the core version of the handler could be limited to the official functionality, you could write custom versions that we either put into the optional custom script folder, which users could simply enable to get the custom version, or we could try to officialize some way to manage content packs, where an adjusted skill handler, or a whole set of them, could become one pack that users can easily install.

Reply:
I mean, we could easily add a feature for it, though I'm guessing you don't want to create features for unofficial code.  It's worth having a discussion about on the discord probably.